src/raftstore/store/fsm/peer.rs

```diff
@@ -668,11 +724,17 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         let (kv_wb, raft_wb, append_res, sync_log) = {
             let mut ctx = ReadyContext::new(&mut self.raft_metrics, &self.trans, pending_count);
             for region_id in self.pending_raft_groups.drain() {
-                if let Some(peer) = self.region_peers.get_mut(&region_id) {
+                if let Some(mut peer) = self.region_peers.remove(&region_id) {
```

Comment:
can we avoid remove and insert again?

Reply:
no, otherwise `self.region_peers` cannot be passed to `handle_rafr_ready_append`